rtrlib/lib/alloc_utils.h

```diff
@@ -34,4 +34,13 @@ void lrtr_free(void *ptr);
 
 void *lrtr_realloc(void *ptr, size_t size);
 
+/**
+ * @brief Duplicates a string
+ * @param[in] string
+ * @returns Duplicated string
+ * @returns NULL if string is NULL
+ * @returns NULL on error
```

Comment:
The main and currently only use case for this function is copying the socket configuration, parts of which may be null. Instead of guarding every copy I decided to write the function accordingly.